### PR TITLE
 [JS OV] Extend `Core` class with `getAvailableDevices`

### PR DESCRIPTION
**Pull request by Aliczi**
### Description:
 Extend `Core` class with `getAvailableDevices` method that returns devices available for inference.
The effective inference of models depends on the hardware platform. Using the `getAvailableDevices` method in the JS API, users will be able to check which platforms are available for running inference.

### Implementation 
- add `get_avaialable_devices()` method on C++ side
- update `ov_node_addon`  TypeScript definition
- create unit test for added functionality using Node.js Test Runner

### CODE REVIEW

File: src/bindings/js/node/lib/addon.ts [hunk 1]
```diff
@@ -173,6 +174,6 @@ export interface NodeAddon {
 }
 
 export default
-    // eslint-disable-next-line @typescript-eslint/no-var-requires
-    require('../bin/ov_node_addon.node') as
-    NodeAddon;
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+require('../bin/ov_node_addon.node') as
+  NodeAddon;
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/lib/addon.ts:179]: Why did this offset change?

File: src/bindings/js/node/tests/basic.test.js [hunk 2]
```diff
@@ -14,6 +15,12 @@ const compiledModel = core.compileModelSync(model, 'CPU');
 const modelLike = [[model],
   [compiledModel]];
 
+it('Core.getAvailableDevices()', () => {
+    if (platform !== 'linux') return;
+    const devices = core.getAvailableDevices();
+    assert.ok(devices.includes('CPU'));
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/basic.test.js:21]:
```suggestion
    if (platform !== 'linux') return;
    
    const devices = core.getAvailableDevices();
    
    assert.ok(devices.includes('CPU'));
```

File: src/bindings/js/node/tests/basic.test.js [hunk 3]
```diff
@@ -14,6 +15,12 @@ const compiledModel = core.compileModelSync(model, 'CPU');
 const modelLike = [[model],
   [compiledModel]];
 
+it('Core.getAvailableDevices()', () => {
+    if (platform !== 'linux') return;
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/basic.test.js:19]: Why is it linux feature only?

File: src/bindings/js/node/lib/addon.ts (hunk 1 above)
Comment (almilosz) [reply src/bindings/js/node/lib/addon.ts]: Fixed

Comment (Aliczi) [reply src/bindings/js/node/lib/addon.ts]: Fixed

File: src/bindings/js/node/tests/basic.test.js (hunk 3 above)
Comment (Aliczi) [reply src/bindings/js/node/tests/basic.test.js]: Test was prepared for a currently supported platform. 
Will be removed if tests shouldn't include limitations of the compilation side.

**Review by vishniakov-nikolai — APPROVED** [review]

**Comment by mlukasze** [comment]
build_jenkins

**Pull request merged**